services/worker-manager/test/api_test.js

```diff
@@ -470,9 +470,22 @@ helper.secrets.mockSuite(testing.suiteName(), ['db'], function(mock, skipping) {
 
     assert.deepStrictEqual(data.workerPools, [], 'Should return an empty array of worker pools');
   });
+  
+  test('get 404 status when worker pool is not present' , async function(){
+       const workerPoolId = 'no/such';
+       try{
+        await helper.workerManager.listWorkersForWorkerPool(workerPoolId);
+       }  catch(err){
+            if(err.code !== 'ResourceNotFound'){
+               throw err;
+            }
+            return;
+          }
+        throw new Error('get of workers from non-existing pool suceeded');   
+  });
```

Comment:
I think this can be more neatly with `assert.rejects`, of which there are a few exampls in this file.